Mathlib/Order/Hom/Set.lean

```diff
@@ -121,10 +121,42 @@ theorem orderIsoOfSurjective_self_symm_apply (b : β) :
 end StrictMono
 
 /-- Two order embeddings on a well-order are equal provided that their ranges are equal. -/
-lemma OrderEmbedding.range_inj [LinearOrder α] [WellFoundedLT α] [PartialOrder β] {f g : α ↪o β} :
+lemma OrderEmbedding.range_inj [LinearOrder α] [WellFoundedLT α] [Preorder β] {f g : α ↪o β} :
     Set.range f = Set.range g ↔ f = g := by
   rw [f.strictMono.range_inj g.strictMono, DFunLike.coe_fn_eq]
 
+namespace OrderIso
+
+instance subsingleton_of_wellFoundedLT [LinearOrder α] [WellFoundedLT α] [Preorder β] :
+    Subsingleton (α ≃o β) := by
+  refine ⟨fun f g ↦ ?_⟩
+  rw [OrderIso.ext_iff, ← coe_toOrderEmbedding, ← coe_toOrderEmbedding, DFunLike.coe_fn_eq,
+    ← OrderEmbedding.range_inj, coe_toOrderEmbedding, coe_toOrderEmbedding, range_eq, range_eq]
+
+instance subsingleton_of_wellFoundedLT' [LinearOrder β] [WellFoundedLT β] [Preorder α] :
+    Subsingleton (α ≃o β) := by
+  refine ⟨fun f g ↦ ?_⟩
+  change f.symm.symm = g.symm.symm
+  rw [Subsingleton.elim f.symm]
```

Comment:
Maybe not useful, but the condition is also true under `[LinearOrder α] [WellFoundedLT β] [Preorder β]`, right? Can you either prove that or add a comment explaining why you didn't prove it?

Reply:
This can be proved by transferring the `WellFoundedLT` instance to `α` using [`RelEmbedding.isWellFounded`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/RelIso/Basic.html#RelEmbedding.isWellFounded). I'll add a comment explaining that (since having four nearly duplicate instances is probably not a good thing).